edge-apps/asset-metadata/static/js/main.js

```diff
@@ -1,130 +1,188 @@
+      }
+      const formatLatitude = (lat) => {
+        const direction = lat > 0 ? 'N' : 'S'
+        return `${formatQuantity(lat)} ${direction}`
+      }
+      const formatLongitude = (lng) => {
+        const direction = lng > 0 ? 'E' : 'W'
+        return `${formatQuantity(lng)} ${direction}`
+      }
+
+      return `${formatLatitude(lat)}, ${formatLongitude(lng)}`
     }
-    const formatLatitude = (lat) => {
-      const direction = lat > 0 ? 'N' : 'S'
-      return `${formatQuantity(lat)} ${direction}`
+
+    // Labels
+    const labelChipContainer = document.querySelector('.label-chip-container')
+    tags.forEach(label => {
+      const chip = document.createElement('div')
+      chip.classList.add('label-chip')
+      chip.innerHTML = `<span class="label-chip-text-label">${label}</span>`
+      labelChipContainer.appendChild(chip)
+    })
+
+    // constant colors
+    const tertiaryColor = '#FFFFFF'
+    const backgroundColor = '#C9CDD0'
+
+    // Theme Selection
+    const theme = screenly.settings.theme ? screenly.settings.theme : 'light'
+
+    // Brand details fetching from settings
+    const primaryColor = (!screenly.settings.screenly_color_accent || screenly.settings.screenly_color_accent.toLowerCase() === '#ffffff') ? '#972eff' : screenly.settings.screenly_color_accent
+
+    let secondaryColor = '#adafbe'
+    if (theme === 'light') {
+      secondaryColor = (!screenly.settings.screenly_color_light || screenly.settings.screenly_color_light.toLowerCase() === '#ffffff') ? '#adafbe' : screenly.settings.screenly_color_light
+    } else if (theme === 'dark') {
+      secondaryColor = (!screenly.settings.screenly_color_dark || screenly.settings.screenly_color_dark.toLowerCase() === '#ffffff') ? '#adafbe' : screenly.settings.screenly_color_dark
     }
-    const formatLongitude = (lng) => {
-      const direction = lng > 0 ? 'E' : 'W'
-      return `${formatQuantity(lng)} ${direction}`
+
+    document.documentElement.style.setProperty('--theme-color-primary', primaryColor)
+    document.documentElement.style.setProperty('--theme-color-secondary', secondaryColor)
+    document.documentElement.style.setProperty('--theme-color-tertiary', tertiaryColor)
+    document.documentElement.style.setProperty('--theme-color-background', backgroundColor)
+
+    // Get the logo image element
+    const imgElement = document.getElementById('brand-logo')
+
+    // Initialize variables
+    let logoUrl = '' // Logo URL
+    let fallbackUrl = '' // Fallback logo if CORS URL fails
+    const defaultLogo = 'static/images/screenly.svg' // Fall back screenly logo
+
+    // Define settings
+    const lightLogo = screenly.settings.screenly_logo_light
+    const darkLogo = screenly.settings.screenly_logo_dark
+
+    // Set logo URLs based on theme
+    if (theme === 'light') {
+      logoUrl = lightLogo
+        ? `${screenly.cors_proxy_url}/${lightLogo}`
+        : `${screenly.cors_proxy_url}/${darkLogo}`
```

Comment:
From my perspective, if the theme is selected light, but light logo is not defined, it should show a placeholder, not a dark logo. But it depends on the documentation. 

If the documentation says, that the theme controls the preferred logo and if it's absent, selects another one, then this logic is fine.

Reply:
For now, we have the primary fallback logo as a screenly one in case any of the brand logos can't fetched, and in case any dark/light logo can't be fetched, it will be displayed as the same brand light/dark logo. Also They can also easily update it in the screenly dashboard brand tab.